pkg/reconciler/awss3source/adapter.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright (c) 2021 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awss3source
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	kr "k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"knative.dev/eventing/pkg/reconciler/source"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/kmeta"
+
+	"github.com/triggermesh/aws-event-sources/pkg/apis/sources/v1alpha1"
+	"github.com/triggermesh/aws-event-sources/pkg/reconciler/common"
+	"github.com/triggermesh/aws-event-sources/pkg/reconciler/common/resource"
+)
+
+const envMessageProcessor = "SQS_MESSAGE_PROCESSOR"
+
+const healthPortName = "health"
+
+// adapterConfig contains properties used to configure the source's adapter.
+// These are automatically populated by envconfig.
+type adapterConfig struct {
+	// Container image
+	// Reuses the adapter from the SQS source.
+	Image string `envconfig:"AWSSQSSOURCE_IMAGE" default:"gcr.io/triggermesh/awssqssource"`
```

Comment:
@antoineco should this be "Image string `envconfig:"AWSS3SOURCE_IMAGE" default:"gcr.io/triggermesh/awss3source"`"?

Reply:
The S3 source consumes bucket notifications over a SQS queue (which we reconcile in the controller if it doesn't exist), therefore the S3 source simply reuses the SQS source's adapter.